src/sas/qtgui/Utilities/Preferences/PreferencesWidget.py

```diff
@@ -56,25 +57,43 @@ def config_value_setter_generator(attr: str, dtype: Optional[Callable] = None):
     :return: a function that takes a single argument, which will be cast to dtype
             and set in config as attr
     """
-    
+
     return functools.partial(set_config_value, attr=attr, dtype=dtype)
 
 
+class QHLine(QFrame):
+    """::CRUFT:: This creates a horizontal line in PyQt5. PyQt6 QFrame has finer shape control"""
+    def __init__(self):
+        super(QHLine, self).__init__()
+        self.setFrameShape(QFrame.HLine)
+        self.setFrameShadow(QFrame.Sunken)
+
 
 class PreferencesWidget(QWidget):
-    """A helper class that bundles all values needed to add a new widget to the preferences panel
-    """
-    # Name that will be added to the PreferencesPanel listWidget
+    """A helper class that bundles all values needed to add a new widget to the preferences panel"""
+    # Default name that will be used in the PreferencesPanel listWidget
     name = None  # type: str
 
     def __init__(self, name: str, default_method: Optional[Callable] = None):
         super(PreferencesWidget, self).__init__()
+        self.parent = None
         self.name = name
         self.resetDefaults = default_method
+        # Create generic layout
         self.verticalLayout = QVBoxLayout()
         self.setLayout(self.verticalLayout)
+        # Child class generates GUI elements
+        self._addAllWidgets()
+        # Push all elements to the top of the window
+        self.verticalLayout.addStretch()
         self.adjustSize()
 
+    def _addAllWidgets(self):
+        """A private psuedo-abstract class that children should override. All widgets should be added here to force
+        elements to the top of the window.
+        """
+        pass
```

Comment:
Since without this override no functionality of this widget exists, why not add a `NotImplementedError` exception thrown when this method is not implemented? Just make it more rigorously abstract.

Reply:
Good idea. I plan to do something similar for restoring defaults.